contracts/tests/MutualConsent.t.sol

```diff
@@ -0,0 +1,457 @@
+        spigot = Spigot(payable(spigot_address));
+
+        escrow_address = address(line.escrow());
+        escrow = Escrow(payable(escrow_address));
+    }
+
+    function setUp() public {
+        // add consent as borrower
+        vm.startPrank(borrower);
+        line.addCredit(dRate, fRate, amount, token, lender);
+        vm.stopPrank();
+
+        _mintAndApprove();
+    }
+
+    /*/////////////////////////////////////////////////////
+                            addCredit
+    /////////////////////////////////////////////////////*/
+
+    function test_addCredit_revoking_invalid_consent_fails() public {
+        // derive the expected consent hash
+        bytes
+            memory invalidMsgData = _generateAddCreditMutualConsentMessageData(
+                ILineOfCredit.addCredit.selector,
+                dRate,
+                fRate,
+                amount,
+                token,
+                makeAddr("randomLender")
+            );
+        emit log_named_bytes("invalid msg data:", invalidMsgData);
+        emit log_named_uint("bytes length", invalidMsgData.length);
+        bytes32 nonExistentHash = _simulateMutualConstentHash(
+            invalidMsgData,
+            borrower
+        );
+        emit log_named_bytes32("invalid msg hash:", nonExistentHash);
+
+        vm.startPrank(borrower);
+        vm.expectRevert(MutualConsent.InvalidConsent.selector);
+        line.revokeConsent(invalidMsgData);
+
+        vm.stopPrank();
+    }
+
+    function test_addCredit_should_fail_revoking_consent_as_malicious_user()
+        public
+    {
+        // should fail revoking consent as different user (with correct data)
+
+        bytes memory msgData = _generateAddCreditMutualConsentMessageData(
+            ILineOfCredit.addCredit.selector,
+            dRate,
+            fRate,
+            amount,
+            token,
+            lender
+        );
+
+        bytes32 expectedHash = _simulateMutualConstentHash(msgData, borrower);
+
+        vm.startPrank(lender);
+        vm.expectRevert(MutualConsent.InvalidConsent.selector);
+        line.revokeConsent(msgData);
+        vm.stopPrank();
+    }
+
+    function test_addCredit_can_revoke_mutualConsent_for_addCredit_as_borrower()
+        public
+    {
+        vm.startPrank(borrower);
+
+        bytes memory msgData = _generateAddCreditMutualConsentMessageData(
+            ILineOfCredit.addCredit.selector,
+            dRate,
+            fRate,
+            amount,
+            token,
+            lender
+        );
+        bytes32 expectedHash = _simulateMutualConstentHash(msgData, borrower);
+
+        // succeed revoking consent
+        vm.expectEmit(true, true, false, true, address(line));
+        emit MutualConsentRevoked(borrower, expectedHash);
+        line.revokeConsent(msgData);
+        vm.stopPrank();
+
+        assertEq(line.mutualConsents(expectedHash), address(0));
+
+        // lender addCredit should create a new id
+        vm.startPrank(lender);
```

Comment:
Can you create this as a second test `revoking_consent_must_delete_consent_hash`

Reply:
Or i guess 2 new tests. 1 check that hash => address is null and 1 check that calling hash after revoking doesnt succeed (technically different stories)

Reply:
@kibagateaux to test for the second condition, the call obviously succeeds, but it creates a new entry in the `MutualConsents` mapping and emits the `MutualConsentRegistered` event, which is the condition for passing the test.